### PR TITLE
Add documentation of existing tests

### LINKED ISSUE #35: Document existing tests
We could add a page in our documentation with all our integration, E2E and maybe unit tests, with a short description for each one. This will let anyone interested see what exactly we test for and maybe raise some questions when some testing is missing.

### PR DESCRIPTION
**Pull request by hug-dev**
Adds a new page in our book to describe the tests that we have on a
high-level view.
Close #35

### CODE REVIEW

File: src/parsec_service/tests/existing_tests.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+## Integration tests on providers
+
+| Name                                                                                          | Description                                                                                                                                                                                                                                                                             |
+|-----------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| [TPM integration tests](https://github.com/parallaxsecond/parsec/tree/master/tests/providers) | These tests initialize a provider structure and perform operation using the `Provide` trait. A software crypto library, like `ring`, can be used to verify the behaviour. [Example with the TPM provider](https://github.com/parallaxsecond/parsec/blob/master/tests/providers/tpm.rs). |
+
```
Comment (ionut-arm) [inline src/parsec_service/tests/existing_tests.md:8]: Sorry to bug you with this, but I'm moving these to the E2E tests 😬  we could do it either way - you change this or  change it after mine gets merged.

File: src/parsec_service/tests/existing_tests.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+and [`config`](https://github.com/parallaxsecond/parsec/tree/master/e2e_tests/tests/config) tests
+
+The
+[`parsec/e2e_tests/provider_cfg`](https://github.com/parallaxsecond/parsec/tree/master/e2e_tests/provider_cfg)
+folder contain the `Dockerfile` and Parsec configuration needed to run Parsec for the corresponding
+tests.
+
+| Name                                                                                                                                     | Description                                                                                                       |
+|------------------------------------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------|
+| [All providers e2e tests](https://github.com/parallaxsecond/parsec/tree/master/e2e_tests/tests/all_providers)                            | End-to-end tests on a service with all providers. Mostly to test core operations.                                 |
```
Comment (ionut-arm) [inline src/parsec_service/tests/existing_tests.md:31]: You could also mention that now we do "cross testing" - either in this line or in another one, where we test that the output of various operations from different providers is accepted by the other providers.

File: src/parsec_service/tests/existing_tests.md (hunk 1 above)
Comment (ionut-arm) [reply src/parsec_service/tests/existing_tests.md:8]: I guess you can just say in the "Normal tests" line below that we also verify Parsec results with 3rd party software implementations of crypto (`ring`, the `rsa` crate...)

**Review by ionut-arm — COMMENTED** [review]
Thanks, this covers our tests quite nicely!

**Comment by hug-dev** [comment]
Thanks for the review! Addressed all of your points. The CI link check will fail on accessing `cross.rs` but will reload once your other PR is merged 🚀

**Review by ionut-arm — APPROVED** [review]
👌

**Pull request merged**